runtime/lua/vim/lsp/diagnostic.lua

```diff
@@ -366,6 +363,7 @@ end
 ---              Structured: { [1] = {...}, [5] = {.... } }
 ---@private
 function M.get_line_diagnostics(bufnr, line_nr, opts, client_id)
+  vim.deprecate('vim.lsp.diagnostic.get_line_diagnostics', 'vim.diagnostic.get', '0.12')
```

Comment:
Is it necessary to deprecate this function if it's marked as `private`?

Reply:
It's a bit grey area:

- It wasn't always marked as `@private` but became private in https://github.com/mfussenegger/neovim/commit/a5bbb932f9094098bd656d3f6be3c58344576709
- It's not prefixed with `_`, users who inspect modules by other means might have seen it

I also still use it myself in nvim-jdtls.
But I accelerated it to 0.12 because of it and I'm not sure if it also warrants the deprecated.txt entry.